addons/web_editor/static/src/js/wysiwyg/widgets/link.js

```diff
@@ -82,115 +118,68 @@ const Link = Widget.extend({
         const allBtnColorPrefixes = /(^|\s+)(bg|text|border)(-[a-z0-9_-]*)?/gi;
         const allBtnClassSuffixes = /(^|\s+)btn(-[a-z0-9_-]*)?/gi;
         const allBtnShapes = /\s*(rounded-circle|flat)\s*/gi;
-        this.data.className = this.data.iniClassName
+        this.state.className = this.state.iniClassName
             .replace(allBtnColorPrefixes, ' ')
             .replace(allBtnClassSuffixes, ' ')
             .replace(allBtnShapes, ' ');
-        this.data.className += ' ' + keptClasses.join(' ');
+        this.state.className += ' ' + keptClasses.join(' ');
         // 'o_submit' class will force anchor to be handled as a button in linkdialog.
-        if (/(?:s_website_form_send|o_submit)/.test(this.data.className)) {
-            this.isButton = true;
+        if (/(?:s_website_form_send|o_submit)/.test(this.state.className)) {
+            this.state.isButton = true;
         }
-
-        this.renderingPromise = new Promise(resolve => this._renderingResolver = resolve);
-    },
+    }
     /**
      * @override
      */
-    start: async function () {
-        for (const option of this._getLinkOptions()) {
-            const $option = $(option);
-            const value = $option.is('input') ? $option.val() : $option.data('value') || option.getAttribute('value');
-            let active = false;
-            if (value) {
-                const subValues = value.split(',');
-                let subActive = true;
-                for (let subValue of subValues) {
-                    const classPrefix = new RegExp('(^|btn-| |btn-outline-|btn-fill-)' + subValue);
-                    subActive = subActive && classPrefix.test(this.data.iniClassName);
-                }
-                active = subActive;
-            } else {
-                active = !this.data.iniClassName
-                         || this.toleratedClasses.some(val => this.data.iniClassName.split(' ').includes(val))
-                         || !this.data.iniClassName.includes('btn-');
-            }
-            this._setSelectOption($option, active);
-        }
-
-        const _super = this._super.bind(this);
+    async start() {
+        this._setSelectOptionFromLink();
 
         this._updateOptionsUI();
 
-        if (this.data.url) {
-            var match = /mailto:(.+)/.exec(this.data.url);
-            this.$('input[name="url"]').val(match ? match[1] : this.data.url);
-            this._onURLInput();
-            this._savedURLInputOnDestroy = false;
-        }
-
-        if (!this.noFocusUrl) {
-            this.focusUrl();
-        }
-
-        return _super(...arguments);
-    },
-    /**
-     * @private
-     */
-    async _widgetRenderAndInsert() {
-        const res = await this._super(...arguments);
-
-        // TODO find a better solution than this during the upcoming refactoring
-        // of the link tools / link dialog.
-        if (this._renderingResolver) {
-            this._renderingResolver();
-        }
```

Comment:
This seems to have been removed without resolving the TODO. Even one `renderingPromise` occurrence is still in the code and is thus probably dead code / recreating the original issue?